src/openrct2/core/Json.hpp

```diff
@@ -11,36 +11,187 @@
+     * @return Copy of the JSON value converted to std::string
+     */
+    std::string GetString(json_t  jsonObj, std::string defaultValue = "");
+
+    /**
+     * Explicit type conversion between a JSON object and a boolean
+     * @param jsonObj JSON object holding the value
+     * @param defaultValue Default value to return if the JSON object is not a string
+     * @return Copy of the JSON value converted to bool
+     */
+    bool GetBoolean(json_t  jsonObj, bool defaultValue = false);
+
+
+    /**
+     * Ensures a given JSON object is an object type
+     * @param jsonObj JSON object
+     * @return The JSON object if it is an object type, or an empty object otherwise
+     */
+    json_t AsObject(json_t  jsonObj);
+
+    /**
+     * Ensures a given JSON object is an array type
+     * @param jsonObj JSON object
+     * @return The JSON object if it is an array type, or an array containing the JSON object otherwise
+     */
+    json_t AsArray(json_t  jsonObj);
+
+
+    /**
+      * Helper function to convert a json object and an initializer list to binary flags
+      * @param T Type to return
+      * @param jsonObj JSON object containing boolean values
+      * @param list List of pairs of keys and bits to enable if that key in the object is true
+      * @return Value with relevant bits flipped
+      */
+    template<typename T>
+    T GetFlags(json_t  jsonObj, std::initializer_list<std::pair<std::string, T>> list)
+    {
+        T flags{};
+        for (const auto& item : list)
+        {
+            if (Json::GetBoolean(jsonObj[item.first]))
+            {
+                flags = static_cast<T>(flags | item.second);
+            }
+        }
+        return flags;
+    }
+
+    /**
+     * Used by the GetFlags function to allow for inverted values
+     */ 
+    enum
+    {
+        // Bit is enabled when key returns true
+        FLAG_NORMAL = 0,
+        // Bit is enabled when key returns false
+        FLAG_INVERTED = 1
+    };
+
+    /**
+      * Helper function to convert a json object and an initializer list to binary flags
+      * @param T Type to return
+      * @param jsonObj JSON object containing boolean values
+      * @param list List of tuples of keys, bits to change and flag type
+      * @return Value with relevant bits flipped
+      * @note FLAG_NORMAL behaves like the other GetFlags function, but FLAG_INVERTED will turn the flag on when false
+      */
+    template<typename T>
+    T GetFlags(json_t  jsonObj, std::initializer_list<std::tuple<std::string, T, uint8_t>> list)
+    {
+        T flags{};
+        for (const auto& item : list)
+        {
+            // true means flag is enabled
+            if (std::get<2>(item) == FLAG_NORMAL)
+            {
+                if (Json::GetBoolean(jsonObj[std::get<0>(item)]))
+                {
+                    flags = static_cast<T>(flags | std::get<1>(item));
+                }
+            }
+            // false means flag is enabled
```

Comment:
```suggestion
            // false means flag is disabled
```

The comment is the same as above, so I'll just assume this is a typo. I think the enum is pretty self explanatory anyway, so I don't see much value in these comments.